File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/SdkworkAiClient.kt
package com.sdkwork.ai

import com.sdkwork.common.core.SdkConfig
import com.sdkwork.ai.http.HttpClient
import com.sdkwork.ai.api.OpenApi
import com.sdkwork.ai.api.ChatApi
import com.sdkwork.ai.api.VideoApi
import com.sdkwork.ai.api.ThreadApi
import com.sdkwork.ai.api.ResponseApi
import com.sdkwork.ai.api.RerankApi
import com.sdkwork.ai.api.MusicApi
import com.sdkwork.ai.api.ModerationApi
import com.sdkwork.ai.api.MessageApi
import com.sdkwork.ai.api.KnowledgeBaseApi
import com.sdkwork.ai.api.ImageApi
import com.sdkwork.ai.api.FileApi
import com.sdkwork.ai.api.EmbeddingApi
import com.sdkwork.ai.api.ContextApi
import com.sdkwork.ai.api.BatchApi
import com.sdkwork.ai.api.AudioApi
import com.sdkwork.ai.api.AssistantApi
import com.sdkwork.ai.api.ModelApi

class SdkworkAiClient {
    private val httpClient: HttpClient

    lateinit var open: OpenApi
    lateinit var chat: ChatApi
    lateinit var video: VideoApi
    lateinit var thread: ThreadApi
    lateinit var response: ResponseApi
    lateinit var rerank: RerankApi
    lateinit var music: MusicApi
    lateinit var moderation: ModerationApi
    lateinit var message: MessageApi
    lateinit var knowledgeBase: KnowledgeBaseApi
    lateinit var image: ImageApi
    lateinit var file: FileApi
    lateinit var embedding: EmbeddingApi
    lateinit var context: ContextApi
    lateinit var batch: BatchApi
    lateinit var audio: AudioApi
    lateinit var assistant: AssistantApi
    lateinit var model: ModelApi

    constructor(baseUrl: String) {
        this.httpClient = HttpClient(baseUrl)
        open = OpenApi(httpClient)
        chat = ChatApi(httpClient)
        video = VideoApi(httpClient)
        thread = ThreadApi(httpClient)
        response = ResponseApi(httpClient)
        rerank = RerankApi(httpClient)
        music = MusicApi(httpClient)
        moderation = ModerationApi(httpClient)
        message = MessageApi(httpClient)
        knowledgeBase = KnowledgeBaseApi(httpClient)
        image = ImageApi(httpClient)
        file = FileApi(httpClient)
        embedding = EmbeddingApi(httpClient)
        context = ContextApi(httpClient)
        batch = BatchApi(httpClient)
        audio = AudioApi(httpClient)
        assistant = AssistantApi(httpClient)
        model = ModelApi(httpClient)
    }

    constructor(config: SdkConfig) {
        this.httpClient = HttpClient(config)
        open = OpenApi(httpClient)
        chat = ChatApi(httpClient)
        video = VideoApi(httpClient)
        thread = ThreadApi(httpClient)
        response = ResponseApi(httpClient)
        rerank = RerankApi(httpClient)
        music = MusicApi(httpClient)
        moderation = ModerationApi(httpClient)
        message = MessageApi(httpClient)
        knowledgeBase = KnowledgeBaseApi(httpClient)
        image = ImageApi(httpClient)
        file = FileApi(httpClient)
        embedding = EmbeddingApi(httpClient)
        context = ContextApi(httpClient)
        batch = BatchApi(httpClient)
        audio = AudioApi(httpClient)
        assistant = AssistantApi(httpClient)
        model = ModelApi(httpClient)
    }

    fun setApiKey(apiKey: String): SdkworkAiClient {
        httpClient.setApiKey(apiKey)
        return this
    }

    fun setAuthToken(token: String): SdkworkAiClient {
        httpClient.setAuthToken(token)
        return this
    }

    fun setAccessToken(token: String): SdkworkAiClient {
        httpClient.setAccessToken(token)
        return this
    }

    fun setHeader(key: String, value: String): SdkworkAiClient {
        httpClient.setHeader(key, value)
        return this
    }
}

File: sdkwork-ai-sdk-swift/Sources/SdkworkAiClient.swift
import Foundation
import SDKworkCommon

public class SdkworkAiClient {
    private let httpClient: HttpClient
    public let open: OpenApi
    public let chat: ChatApi
    public let video: VideoApi
    public let thread: ThreadApi
    public let response: ResponseApi
    public let rerank: RerankApi
    public let music: MusicApi
    public let moderation: ModerationApi
    public let message: MessageApi
    public let knowledgeBase: KnowledgeBaseApi
    public let image: ImageApi
    public let file: FileApi
    public let embedding: EmbeddingApi
    public let context: ContextApi
    public let batch: BatchApi
    public let audio: AudioApi
    public let assistant: AssistantApi
    public let model: ModelApi

    public init(baseURL: String) {
        self.httpClient = HttpClient(baseURL: baseURL)
        self.open = OpenApi(client: httpClient)
        self.chat = ChatApi(client: httpClient)
        self.video = VideoApi(client: httpClient)
        self.thread = ThreadApi(client: httpClient)
        self.response = ResponseApi(client: httpClient)
        self.rerank = RerankApi(client: httpClient)
        self.music = MusicApi(client: httpClient)
        self.moderation = ModerationApi(client: httpClient)
        self.message = MessageApi(client: httpClient)
        self.knowledgeBase = KnowledgeBaseApi(client: httpClient)
        self.image = ImageApi(client: httpClient)
        self.file = FileApi(client: httpClient)
        self.embedding = EmbeddingApi(client: httpClient)
        self.context = ContextApi(client: httpClient)
        self.batch = BatchApi(client: httpClient)
        self.audio = AudioApi(client: httpClient)
        self.assistant = AssistantApi(client: httpClient)
        self.model = ModelApi(client: httpClient)
    }

    public init(config: SdkConfig) {
        self.httpClient = HttpClient(config: config)
        self.open = OpenApi(client: httpClient)
        self.chat = ChatApi(client: httpClient)
        self.video = VideoApi(client: httpClient)
        self.thread = ThreadApi(client: httpClient)
        self.response = ResponseApi(client: httpClient)
        self.rerank = RerankApi(client: httpClient)
        self.music = MusicApi(client: httpClient)
        self.moderation = ModerationApi(client: httpClient)
        self.message = MessageApi(client: httpClient)
        self.knowledgeBase = KnowledgeBaseApi(client: httpClient)
        self.image = ImageApi(client: httpClient)
        self.file = FileApi(client: httpClient)
        self.embedding = EmbeddingApi(client: httpClient)
        self.context = ContextApi(client: httpClient)
        self.batch = BatchApi(client: httpClient)
        self.audio = AudioApi(client: httpClient)
        self.assistant = AssistantApi(client: httpClient)
        self.model = ModelApi(client: httpClient)
    }

    public func setApiKey(_ apiKey: String) -> SdkworkAiClient {
        httpClient.setApiKey(apiKey)
        return self
    }

    public func setAuthToken(_ token: String) -> SdkworkAiClient {
        httpClient.setAuthToken(token)
        return self
    }

    public func setAccessToken(_ token: String) -> SdkworkAiClient {
        httpClient.setAccessToken(token)
        return self
    }

    public func setHeader(_ key: String, value: String) -> SdkworkAiClient {
        httpClient.setHeader(key, value: value)
        return self
    }
}
